tests/test_priors.py

```diff
@@ -115,9 +129,59 @@ def testUniform(self):
             m.x = np.random.randn(1)[0]
             p2 = m.compute_log_prior()
 
-            # prior should no be the same because a transformation has been applied.
+            # prior should not be the same because a transformation has been applied.
             self.assertTrue(p1 != p2)
 
 
+def mc_moments(prior, size=(10000, 1000)):
+    np.random.seed(1)
+    x = prior.sample(size)
+    assert x.shape == size
+    mean = x.mean()
+    var = x.var()
+    return mean, var
+
+def gaussian_moments(prior):
+    return prior.mu, prior.var
+
+def exponential_moments(prior):
+    return 1 / prior.rate, prior.rate ** (-2)
+
+def lognormal_moments(prior):
+    mu, var = prior.mu, prior.var
+    return np.exp(mu + var / 2), (np.exp(var) - 1) * np.exp(2 * mu + var)
+
+def gamma_moments(prior):
+    return prior.shape * prior.scale, prior.shape * prior.scale ** 2
+
+def laplace_moments(prior):
+    return prior.mu, 2 * prior.sigma ** 2
+
+def beta_moments(prior):
+    a, b = prior.a, prior.b
+    return a / (a + b), (a * b) / ((a + b)**2 * (a + b + 1))
+
+def uniform_moments(prior):
+    a, b = prior.lower, prior.upper
+    return (a + b) / 2, (b - a)**2 / 12
+
+@pytest.mark.parametrize("args", [
+    ("Exponential", [1.3]),
+    ("Gaussian", [-2.5, 3.4]),
+    ("LogNormal", [-2.5, 1.4]),
+    ("Gamma", [1.5, 0.7]),
+    ("Laplace", [-2.5, 3.4]),
+    ("Beta", [3.6, 0.4]),
+    ("Uniform", [5.4, 8.9]),
+    ])
+def test_moments(args):
+    classname, params = args
+    cls = eval("gpflow.priors.{}".format(classname))
+    prior = cls(*params)
+    moments_func = eval("{}_moments".format(cls.__name__.lower()))
+    rtol = 5e-3 if classname == "LogNormal" else 1e-3
+    assert_allclose(np.array(moments_func(prior)).flatten(), mc_moments(prior), rtol=rtol,
```

Comment:
No, it doesn't. I thought it would, but it only says "test_moments[args1,2,3...]".
I can make mc_moments() return arrays instead of scalars if you prefer it there. (The discrepancy is due to the np.atleast_1d in all the prior `__init__`s)